module/modmanager/manager.go

```diff
@@ -115,12 +125,80 @@ func (mgr *Manager) Add(ctx context.Context, cfg config.Module) error {
 		return errors.WithMessage(err, "error while waiting for module to be ready "+mod.name)
 	}
 
-	mod.registerResources(mgr, mgr.logger)
+	// Do not re-register resources if module is being reconfigured.
+	if !reconfigure {
+		mod.registerResources(mgr, mgr.logger)
+	}
 
 	success = true
 	return nil
 }
 
+// Reconfigure reconfigures an existing resource module.
+func (mgr *Manager) Reconfigure(ctx context.Context, cfg config.Module) error {
+	module, exists := mgr.modules[cfg.Name]
+	if !exists {
+		return errors.Errorf("cannot reconfigure module %s as it does not exist", cfg.Name)
+	}
+	handledResources := module.resources
+
+	if err := mgr.remove(cfg.Name, true); err != nil {
+		return err
+	}
+
+	if err := mgr.add(ctx, cfg, true); err != nil {
+		return err
+	}
+
+	// add old module process' resources to new module.
+	for _, res := range handledResources {
+		if _, err := mgr.AddResource(ctx, res.cfg, res.deps); err != nil {
+			return errors.WithMessage(err, "error while re-adding resource to module "+cfg.Name)
+		}
+	}
+	return nil
+}
+
+// Remove removes and stops an existing resource module.
+func (mgr *Manager) Remove(modName string) error {
+	return mgr.remove(modName, false)
+}
+
+func (mgr *Manager) remove(modName string, reconfigure bool) error {
+	mgr.mu.Lock()
+	defer mgr.mu.Unlock()
+
+	mod, exists := mgr.modules[modName]
+	if !exists {
+		return errors.Errorf("cannot remove module %s as it does not exist", modName)
+	}
+
+	if err := mod.stopProcess(); err != nil {
+		return errors.WithMessage(err, "error while stopping module "+modName)
+	}
+
+	// Do not close connection or deregister resources if module is being reconfigured.
```

Comment:
why not deregister? there's no guarantee that the updated module will serve the same models

Reply:
good point; I don't think we need special logic for resource registration in reconfiguration.

Reply:
what would happen if we tried to reconfigure a module that no longer serves the model that the previous module served? I assume we try to build it once but then it'd get orphaned.

I wonder if we actually need to readd the resources at a higher level - basically flag that a rebuild is needed as part of the reconfiguration process but with the awareness that it shouldn't be a reconfigure request